core/src/main/java/hudson/slaves/JNLPLauncher.java

```diff
@@ -68,12 +70,22 @@
 
     @Nonnull
     private RemotingWorkDirSettings workDirSettings;
-    
+
+    /**
+     * Constructor.
+     * @param tunnel Tunnel settings
+     * @param vmargs JVM arguments
+     * @param workDirSettings Settings for Work Directory management in Remoting.
+     *                        If {@code null}, {@link RemotingWorkDirSettings#getEnabledDefaults()}
+     *                        will be used to enable work directories by default in new agents.
+     * @since 2.68
+     */
     @DataBoundConstructor
-    public JNLPLauncher(@CheckForNull String tunnel, @CheckForNull String vmargs, @Nonnull RemotingWorkDirSettings workDirSettings) {
+    public JNLPLauncher(@CheckForNull String tunnel, @CheckForNull String vmargs, @Nullable RemotingWorkDirSettings workDirSettings) {
```

Comment:
Why the mix of `Nullable` and `CheckForNull`?

Reply:
No specific reason. I just wanted to indicate that `null` is not something we would really want to see. Can change it